src/game.py

```diff
@@ -0,0 +1,301 @@
+        self.__player_manager.login()
+
+        # Add the queued local players to the game
+        self.__player_manager.connect_queued_players()
+
+        game_state: dict | None = self.__wait_for_full_lobby()
+
+        if not game_state:
+            # the game was interrupted
+            return
+
+        self.__num_turns = game_state["num_turns"]
+        self.__max_players = game_state["num_players"]
+        self.__num_rounds = game_state["num_rounds"]
+
+        # add all remote players and observers
+        self.__player_manager.add_remote_players(game_state["players"])
+        self.__player_manager.add_remote_players(game_state["observers"])
+
+        # start all player instances
+        self.__player_manager.start_players()
+
+        # set the player win counts
+        for idx in self.__active_players.keys():
+            self.__player_wins[idx] = 0
+
+        self.__start_next_round()
+
+        # output the game info to console
+        print(self)
+
+    def __start_next_round(self) -> None:
+        # start the next round
+        self.__next_round = False
+
+        game_state: dict = self.__shadow_client.get_game_state()
+
+        self.__current_round = game_state["current_round"]
+
+        client_map: dict = self.__shadow_client.get_map()
+
+        # register round start for every player
+        for player in self.__active_players.values():
+            player.register_round()
+
+        # initialize the game map (now adds tanks to players & game_map too)
+        self.game_map = Map(client_map, game_state, self.__active_players, self.__current_turn)
+
+        # pass Map reference to players
+        for player in self.__active_players.values():
+            player.add_map(self.game_map)
+
+    def __start_next_turn(self) -> None:
+        # start the next turn
+        game_state = self.__shadow_client.get_game_state()
+
+        self.__current_turn[0] = game_state["current_turn"]
+        self.__current_player_idx[0] = game_state["current_player_idx"]
+
+        print()
+        if self.__current_player_idx[0] != 0:
+            self.__current_player = self.__active_players[self.__current_player_idx[0]]
+            # Reset current player attacks
+            self.__current_player.register_turn()
+            print(f"Current turn: {self.__current_turn[0]}, "
+                  f"current player: {self.__current_player.player_name}")
+        else:
+            print(f"Current turn: {self.__current_turn[0]}")
+            self.__current_player = None
+
+        if self.game_map:
+            self.game_map.update_turn(game_state)
+
+        if game_state["finished"]:
+            self.__winner = game_state["winner"]
+            if self.__winner is None:
+                self.__game_is_draw = True
+            else:
+                self.__player_wins[self.__winner] += 1
+            self.__print_round_winner()
+
+            if not self.__is_full or game_state["current_round"] == self.__num_rounds:
```

Comment:
for full game num_rounds = 1, you can check only this

Reply:
Yes, you are right, thank you.